main.tf

```
@@ -1,5 +1,6 @@
 locals {
-  bucket_name = var.bucket_name != null && var.bucket_name != "" ? var.bucket_name : module.this.id
+  bucket_name         = var.bucket_name != null && var.bucket_name != "" ? var.bucket_name : module.this.id
+  replication_enabled = length(var.replication_rules) > 0
 }
```

Please use the existing locals block instead of creating a new one.

Also, please drop the `this` prefix to be consistent with `local.bucket_name`

```suggestion
bucket_arn = "arn:${data.aws_partition.current.partition}:s3:::${join("", aws_s3_bucket.default.*.id)}"
}
```